pkg/collector/worker/worker_test.go

```diff
@@ -331,14 +331,14 @@ func TestWorkerUtilizationExpvars(t *testing.T) {
 
 	// No tasks should equal no utilization
 
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(200 * time.Millisecond)
 	require.InDelta(t, getWorkerUtilizationExpvar(t, "worker_2"), 0, 0)
 
 	// High util checks should be reflected in expvars
 
 	pendingChecksChan <- blockingCheck
 
-	time.Sleep(2000 * time.Millisecond)
+	time.Sleep(1200 * time.Millisecond)
```

Comment:
This change is the one I am least confident in. Locally and in CI, I did not get any failures after several dozens of attempts ([CI Visibility](https://app.datadoghq.com/ci/test/runs?query=test_level%3Atest%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fdatadog-agent%20%40test.service%3Adatadog-agent%20%40git.branch%3Akserrania%2Ftest-optim%20%40test.name%3ATestWorkerUtilizationExpvars&agg_m=count&agg_m_source=base&agg_t=count&citest_explorer_sort=time%2Cdesc&fromUser=false&index=citest&mode=sliding&saved-view-id=2721651&start=1743775027596&end=1744379827596&paused=false)), but I did get flakes with lower values like 1s.

Reply:
Would it make sense to replace with `EventuallyWithT`?

Reply:
Good idea, I'll push a commit to use it.